Keas.Mvc/ClientApp/components/Access/AccessDetails.tsx

```diff
@@ -1,129 +1,151 @@
 ﻿import * as React from 'react';
+import { useContext, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
 import { toast } from 'react-toastify';
 import { Button } from 'reactstrap';
 import { Context } from '../../Context';
 import { IAccess } from '../../models/Access';
 import SearchTags from '../Tags/SearchTags';
 import AccessAssignmentContainer from './AccessAssignmentContainer';
 import HistoryContainer from '../History/HistoryContainer';
+import EditAccess from './EditAccess';
 
 interface IProps {
   goBack: () => void;
   modal: boolean;
   closeModal: () => void;
   selectedAccess: IAccess;
-  openEditModal: (access: IAccess) => void;
+  editAccess: (accaess: IAccess) => void;
   openDeleteModal: (access: IAccess) => void;
   updateSelectedAccess: (access: IAccess, id?: number) => void;
 }
 
-export default class AccessDetails extends React.Component<IProps, {}> {
-  public static contextType = Context;
-  public context!: React.ContextType<typeof Context>;
+const AccessDetails = (props: IProps) => {
+  const context = useContext(Context);
+  const history = useHistory();
+  const [shouldOpenEditModal, setEditModal] = useState<boolean>(false);
```

Comment:
as a convention the useState should be `variableName` and `setVariableName` so it forms an easy to find pair. In this example that would mean naming them `const [shouldOpenEditModal, setShouldOpenEditModal]`